src/Symfony/Component/Form/Extension/Core/Type/ChoiceType.php

```diff
@@ -62,6 +62,11 @@ public function __construct(ChoiceListFactoryInterface $choiceListFactory = null
             )
         );
 
+        if (null !== $translator && !$translator instanceof TranslatorInterface) {
+            throw new \TypeError(sprintf('Argument 2 passed to "%s()" must be han instance of "%s", "%s" given.', __METHOD__, TranslatorInterface::class, \is_object($translator) ? \get_class($translator) : \gettype($translator)));
```

Comment:
```suggestion
            throw new \TypeError(sprintf('Argument 2 passed to "%s()" must be an instance of "%s", "%s" given.', __METHOD__, TranslatorInterface::class, \is_object($translator) ? \get_class($translator) : \gettype($translator)));
```

Reply:
Thanks for the suggestion. The parameter type hint suggested by @xabbuh in the next comment feels to be a cleaner solution, I will go for it.